lib/cassia/response_handlers/discover_all.rb

```diff
@@ -0,0 +1,41 @@
+module Cassia
+  module ResponseHandlers
+    class DiscoverAll
+      def initialize(access_controller, router: , device_mac: )
+        @access_controller = access_controller
+        @router = router
+        @device_mac = device_mac
+      end
+
+      def handle(response)
+        if response.success?
+          handle_success(response)
+        else
+          handle_failure(response)
+        end
+        response.success?
+      end
+
+      private
+
+      def handle_success(response)
+        device = @router.connected_devices.detect {|device| device.mac == @device_mac}
+        response.body.each do |service|
+          new_service = Service.new(uuid: service["uuid"], primary: service["primary"], handle: service["handle"])
+          if !(service["characteristics"].nil?)
```

Comment:
instead of doing `!(service["characteristics"].nil?)` you can do `service["characteristics"]`. This will evaluate to true if there is a characteristic